### ISSUE TITLE
Curses.h and stdbool.h compatibility

### ISSUE THREAD

**Issue by CraftCoderr** [Bill-Gray/PDCurses#101]
Why curses.h haven't 
`#ifdef PDC_USE_STD_BOOL
#include "stdbool.h"
#else
   ...
#endif` 
condition to provide stdbool.h compatibility?

**Comment by GitMensch** [comment]
Why should it? I guess to allow conditional use if standard definitions from the compiler.

I see no issues in doing so, can you provide a PR for this change?

**Comment by Bill-Gray** [comment]
I'd not encountered the problem before,  but now that you mention it... compiling a truly minimal example
```
#include <stdbool.h>
#include <curses.h>
```
gets you
```
In file included from z1.c:1:0:
../curses.h:93:23: error: two or more data types in declaration specifiers
 typedef unsigned char bool;    /* PDCurses Boolean type */
                       ^
In file included from z1.c:2:0:
../curses.h:93:1: warning: useless type name in empty declaration
 typedef unsigned char bool;    /* PDCurses Boolean type */
```
@CraftCoderr,  I assume this is the issue you'd like to see fixed? 
I _think_ this can be avoided reliably by putting the above typedef for `bool` between a check to see if  <stdbool.h> has already been included :
```
#ifndef __bool_true_false_are_defined
    typedef unsigned char bool;    /* PDCurses Boolean type */
#endif
```
If you confirm that's the actual issue,  and that the above fixes it,  I'd definitely take that (and it should go upstream to @wcmbrine's PDCurses.)

**Comment by wmcbrine** [comment]
Reliably, maybe not. Consider that `__bool_true_false_are_defined` may be on the way out:

https://wg21.cmeerw.net/lwg/issue2155

(They offer no replacement, and say, erroneously, "... the usefulness of the transition tool has faded long ago.")

But for now, maybe it's the best approach?

**Comment by wmcbrine** [comment]
See here for more: https://github.com/wmcbrine/PDCurses/pull/10

**Comment by CraftCoderr** [comment]
How to properly define C standard version with CMake?

**Comment by wmcbrine** [comment]
I don't see how CMake enters into it? It's defined by the compiler.

**Comment by GitMensch** [comment]
Fixed by following @wmcbrine's lead with 6bc4879c31ab0078f89e41096ec6175b026d1932.
